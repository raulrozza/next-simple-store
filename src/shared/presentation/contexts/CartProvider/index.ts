export { default as useAddToCart } from './hooks/useAddToCart';
