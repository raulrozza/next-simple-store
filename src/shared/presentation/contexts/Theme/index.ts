export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
