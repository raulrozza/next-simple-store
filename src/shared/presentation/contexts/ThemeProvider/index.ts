export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
