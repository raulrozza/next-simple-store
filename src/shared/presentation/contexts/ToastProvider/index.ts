export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
