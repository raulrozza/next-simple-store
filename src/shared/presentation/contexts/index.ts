export * from './CartProvider';
export * from './ThemeProvider';
