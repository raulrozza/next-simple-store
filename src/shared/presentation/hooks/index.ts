import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
