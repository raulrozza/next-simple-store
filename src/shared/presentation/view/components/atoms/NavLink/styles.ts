import styled, { css } from 'styled-components';

export const Anchor = styled.a`
    display: flex;
    align-items: center;
    width: 100%;

    ${({ theme }) => css`
        gap: ${theme.layout.spacing(1)};
        padding: ${theme.layout.spacing(1, 0.5)};
        color: ${theme.palette.gray['0']};
        text-decoration: none;
        font-size: ${theme.typography.sizes.md};
        font-family: ${theme.typography.family.title};

        &:hover {
            color: ${theme.palette.primary['0']};
        }
    `}
`;
