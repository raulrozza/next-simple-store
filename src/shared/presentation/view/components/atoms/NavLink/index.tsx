import { FC } from 'react';

import Link from 'next/link';
import { AiFillHome, AiFillShopping } from 'react-icons/ai';
import { BsFillPeopleFill } from 'react-icons/bs';
import { FaListAlt } from 'react-icons/fa';

import { Anchor } from './styles';

const ICONS = {
    home: AiFillHome,
    list: FaListAlt,
    people: BsFillPeopleFill,
    shopping: AiFillShopping,
};

interface NavLinkProps {
    href: string;
    icon: keyof typeof ICONS;
    active: boolean;
}

const NavLink: FC<NavLinkProps> = ({ href, icon, active, children }) => {
    const Icon = ICONS[icon];

    return (
        <Link href={href} passHref>
            <Anchor aria-selected={active} active={active}>
                <Icon aria-label={`${icon} icon`} />

                {children}
            </Anchor>
        </Link>
    );
};

export default NavLink;
