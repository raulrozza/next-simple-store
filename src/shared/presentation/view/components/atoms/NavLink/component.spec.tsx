import { render, screen } from '@testing-library/react';

import theme from '@/config/theme';
import { ThemeProvider } from '@/shared/presentation/contexts';

import NavLink from './index';

describe('NavLink', () => {
    it('should render the component correctly displaying the correct icon and text', () => {
        const view = render(
            <NavLink href="/" icon="home" active={false}>
                Home
            </NavLink>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Home')).toBeInTheDocument();
        expect(screen.getByLabelText('home icon')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });

    it('should render the a primary color and data selected active if the active prop is true', () => {
        const view = render(
            <NavLink href="/" icon="home" active={true}>
                Home
            </NavLink>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Home')).toHaveStyle({
            color: theme.palette.primary['900'],
        });
        expect(screen.getByText('Home')).toHaveAttribute(
            'aria-selected',
            'true',
        );
        expect(view).toMatchSnapshot();
    });
});
