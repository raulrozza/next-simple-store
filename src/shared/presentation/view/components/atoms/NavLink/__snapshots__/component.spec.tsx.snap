// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should render the a primary color and data selected active if the active prop is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
  color: #3D879D;
}

.c0:hover {
  color: #375758;
}

<body>
    <div>
      <a
        aria-selected="true"
        class="c0"
        href="/"
      >
        <svg
          aria-label="home icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
          />
        </svg>
        Home
      </a>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
  color: #3D879D;
}

.c0:hover {
  color: #375758;
}

<div>
    <a
      aria-selected="true"
      class="c0"
      href="/"
    >
      <svg
        aria-label="home icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
        />
      </svg>
      Home
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavLink should render the component correctly displaying the correct icon and text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
}

.c0:hover {
  color: #375758;
}

<body>
    <div>
      <a
        aria-selected="false"
        class="c0"
        href="/"
      >
        <svg
          aria-label="home icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
          />
        </svg>
        Home
      </a>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
}

.c0:hover {
  color: #375758;
}

<div>
    <a
      aria-selected="false"
      class="c0"
      href="/"
    >
      <svg
        aria-label="home icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
        />
      </svg>
      Home
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
